### PR TITLE
Copilot docs overhaul

### PR DESCRIPTION
**Pull request by mnicstruwig**
Refreshes existing documentation, and adds in new sections about in-line artifacts and citations.

### CODE REVIEW

File: content/terminal/openbb-copilot.md [hunk 1]
```diff
@@ -18,202 +18,318 @@ import ReactPlayer from 'react-player'
...
 If you'd like to clear the history of the current conversation, you can do so by
-clicking on the trashcan icon in the OpenBB Copilot chatbox in the upper right
-corner. It is usually a good idea to clear the chat history when you have a new
-question that is unrelated to your current conversation.
+clicking on the trashcan icon in the upper right corner of the OpenBB Copilot
+window. It is **highly recommended** to clear the chat history when you have a
+new question that is unrelated to your current conversation, to prevent
+confusing OpenBB Copilot:
 
-| ![clear chat button openbb copilot](https://openbb-assets.s3.amazonaws.com/docs/copilot/clear_chat_button.jpg) | ![cleared chat openbb copilot](https://openbb-assets.s3.amazonaws.com/docs/copilot/cleared_chat.jpg)
```
Comment (mnicstruwig) [inline content/terminal/openbb-copilot.md:63]: Remove from S3 after merge

File: content/terminal/openbb-copilot.md [hunk 2]
```diff
@@ -18,202 +18,318 @@ import ReactPlayer from 'react-player'
...
+
+Sometimes, you might want OpenBB Copilot to analyze data from only a specific
+widget (or set of widgets) on your dashboard. For example, if you want to deep dive
+into an earnings transcript, and ignore all other widgets on your dashboard, you can
+do that by clicking on the "Add widgets as context" button on the widgets you want OpenBB
+Copilot to use. Once selected, OpenBB Copilot will only access data from those widgets,
+ignoring all others:
 
 <div style={{display: 'flex', justifyContent: 'center'}}>
-<img src="https://openbb-assets.s3.amazonaws.com/docs/copilot/add_specific_widget_context.jpg" alt="add widget context openbb copilot" width="80%" height="80%" />
```
Comment (mnicstruwig) [inline content/terminal/openbb-copilot.md:88]: Remove from S3 after merge

File: content/terminal/openbb-copilot.md [hunk 3]
```diff
@@ -18,202 +18,318 @@ import ReactPlayer from 'react-player'
...
+<img src="https://openbb-assets.s3.amazonaws.com/docs/copilot/add_specific_widget_context.png" alt="add widget context openbb copilot" width="80%" height="80%" />
 </div>
 
+The OpenBB Copilot window will also show you which widgets are currently selected as explicitly-added context:
+
 <div style={{display: 'flex', justifyContent: 'center'}}>
-<img src="https://openbb-assets.s3.amazonaws.com/docs/copilot/aapl_earnings_transcript_chat_example.jpg" alt="query widget context openbb copilot" width="80%" height="80%" />
```
Comment (mnicstruwig) [inline content/terminal/openbb-copilot.md:92]: Remove from S3 after merge

File: content/terminal/openbb-copilot.md [hunk 4]
```diff
@@ -18,202 +18,318 @@ import ReactPlayer from 'react-player'
...
-For example, if you ask for a summary of the latest news about Apple, OpenBB
-Copilot will retrieve the Company News widget data on your dashboard (provided
-the widget has been added to the dashboard) to provide an answer.
+<div style={{display: 'flex', justifyContent: 'center'}}>
+<img src="https://openbb-assets.s3.amazonaws.com/docs/copilot/reasoning_steps.png" alt="reasoning steps openbb copilot" width="50%" height="50%" />
+</div>
 
+OpenBB Copilot also provides a citation for the data source after its response. When you hover over the citation, it will highlight the widget on the dashboard that provided the data:
 <div style={{display: 'flex', justifyContent: 'center'}}>
-<img src="https://openbb-assets.s3.amazonaws.com/docs/copilot/aapl_+news_chat_example.jpg" alt="aapl news openbb copilot" width="80%" height="80%" />
```
Comment (mnicstruwig) [inline content/terminal/openbb-copilot.md:104]: Remove from S3 after merge

File: content/terminal/openbb-copilot.md [hunk 5]
```diff
@@ -18,202 +18,318 @@ import ReactPlayer from 'react-player'
...
+the relevant widgets on your dashboard.
+
+:::
+
 
 ### Using your own files
 
 OpenBB Copilot can also analyze and answer questions using files you provide, such as PDF, CSV, and XLSX files.
 
-To add a file to OpenBB Copilot, first [add it to the currently-active dashboard](https://docs.openbb.co/pro/upload-files) or click the paper clip icon.
```
Comment (mnicstruwig) [inline content/terminal/openbb-copilot.md:118]: Remove from S3 after merge

File: content/terminal/openbb-copilot.md [hunk 6]
```diff
@@ -18,202 +18,318 @@ import ReactPlayer from 'react-player'
...
+To add a file to OpenBB Copilot, first [upload it and add it as a widget to the currently-active dashboard](https://docs.openbb.co/pro/upload-files).
 
+The file can then be used by OpenBB Copilot to answer questions (either by adding the file widget to the [context explicitly](#analyze-specific-widgets-only), or by allowing OpenBB Copilot to [automatically retrieve the data](#automatically-use-data-from-the-dashboard) if it is relevant to your query).
 
 <div style={{display: 'flex', justifyContent: 'center'}}>
- <img src="https://openbb-assets.s3.amazonaws.com/docs/copilot/add_custom_data_chat_example.jpg" alt="add custom data openbb copilot" width="80%" height="80%" />
```
Comment (mnicstruwig) [inline content/terminal/openbb-copilot.md:122]: Remove from S3 after merge

File: content/terminal/openbb-copilot.md [hunk 7]
```diff
@@ -18,202 +18,318 @@ import ReactPlayer from 'react-player'
...
- </div>
-
- <div style={{display: 'flex', justifyContent: 'center'}}>
- <img src="https://openbb-assets.s3.amazonaws.com/docs/copilot/custom_data_chat_example.jpg" alt="query custom data openbb copilot" width="80%" height="80%" />
```
Comment (mnicstruwig) [inline content/terminal/openbb-copilot.md:126]: Remove from S3 after merge

File: content/terminal/openbb-copilot.md [hunk 8]
```diff
@@ -18,202 +18,318 @@ import ReactPlayer from 'react-player'
...
+<img src="https://openbb-assets.s3.amazonaws.com/docs/copilot/custom_pdf_chat_example.jpg" alt="custom pdf copilot chat" width="100%" height="100%" />
+</div>
 
-If OpenBB Copilot uses your file widget to answer any part of you query, it will show citations after its answer.  
+Alternatively, you can also upload files directly to OpenBB Copilot by clicking
+on the paper clip icon in the OpenBB Copilot window. Files uploaded directly to
+OpenBB Copilot in this manner will **not** be persisted between sessions:
 
 <div style={{display: 'flex', justifyContent: 'center'}}>
-<img src="https://openbb-assets.s3.amazonaws.com/docs/copilot/custom_pdf_chat_example.jpg" alt="custom pdf copilot chat" width="80%" height="80%" />
```
Comment (mnicstruwig) [inline content/terminal/openbb-copilot.md:134]: Remove from S3 after merge

File: content/terminal/openbb-copilot.md [hunk 9]
```diff
@@ -18,202 +18,318 @@ import ReactPlayer from 'react-player'
...
-**Querying the OpenBB API**
+:::
 
-For example, you can ask OpenBB Copilot to fetch the latest stock price for a specific company, or to provide a summary of the latest news for a particular stock without having to add a widget to your dashboard.
+For example, you can ask OpenBB Copilot to fetch the latest stock price for a
+specific company, or to provide a summary of the latest news for a particular
+stock without having to add a widget to your dashboard.
 
 <div style={{display: 'flex', justifyContent: 'center'}}>
-<img src="https://openbb-assets.s3.amazonaws.com/docs/copilot/news_openbb_api_chat_example.jpg" alt="openbb copilot api" width="50%" height="50%" />
```
Comment (mnicstruwig) [inline content/terminal/openbb-copilot.md:161]: Remove from S3 after merge

File: content/terminal/openbb-copilot.md [hunk 10]
```diff
@@ -18,202 +18,318 @@ import ReactPlayer from 'react-player'
...
+- Vertical bar chart
+- Scatter plot
 
-To search the web, the user needs to be explicit in their query by adding the `@web` keyword in the query.
+To produce a chart or a table, mention the output that you want in your query.
+For example, to produce a chart of the latest stock price for Apple, you can
+ask OpenBB Copilot to "Create a chart of the latest stock price for Apple".
 
 <div style={{display: 'flex', justifyContent: 'center'}}>
-<img src="https://openbb-assets.s3.amazonaws.com/docs/copilot/web_search_chat_example.jpg" alt="web search openbb copilot" width="80%" height="80%" />
```
Comment (mnicstruwig) [inline content/terminal/openbb-copilot.md:173]: Remove from S3 after merge

File: content/terminal/openbb-copilot.md [hunk 11]
```diff
@@ -18,202 +18,318 @@ import ReactPlayer from 'react-player'
...
 
-When querying OpenBB Copilot, it is important to understand which data OpenBB
-Copilot used to answer your query. As a result, OpenBB Copilot will always cite the data
-source it referenced when responding to your query.
+### Converting artifacts to widgets
+Every artifact that OpenBB Copilot generates can be converted to a widget. To do
+this, click the "Create widget from text" or "Create widget from table" button:
 
 <div style={{display: 'flex', justifyContent: 'center'}}>
-<img src="https://openbb-assets.s3.amazonaws.com/docs/copilot/citation_chat_example.jpg" alt="citations openbb copilot" width="80%" height="80%" />
```
Comment (mnicstruwig) [inline content/terminal/openbb-copilot.md:183]: Remove from S3 after merge

File: content/terminal/openbb-copilot.md [hunk 12]
```diff
@@ -18,202 +18,318 @@ import ReactPlayer from 'react-player'
...
 
-A special kind of status update is the result artifact which has the final data used to answer the query. It either be a text or a table.
-The contents of the artifact are produced by the reasoning of the OpenBB Copilot.
+OpenBB Copilot has the ability to search the web. 
+
+To search the web, use the `@web` keyword in your query to tell OpenBB Copilot
+to search the web for information. For example:
 
 <div style={{display: 'flex', justifyContent: 'center'}}>
-<img src="https://openbb-assets.s3.amazonaws.com/docs/copilot/artifact_chat_example.jpg" alt="result artifact openbb copilot" width="80%" height="80%" />
```
Comment (mnicstruwig) [inline content/terminal/openbb-copilot.md:195]: Remove from S3 after merge

File: content/terminal/openbb-copilot.md [hunk 13]
```diff
@@ -18,202 +18,318 @@ import ReactPlayer from 'react-player'
...
-To ensure transparency and extend the analysis, OpenBB Copilot will also return intermediate result artifacts that were used as reasoning steps to arrive to the final artifact and thus the answer.
+## Reasoning Steps
+As OpenBB Copilot goes through the process of answering your query, it will show
+you the reasoning steps it took to arrive at the answer.  These reasoning steps
+appear above Copilot's responses. Reasoning steps can be expanded to see more
+information about the intermediate steps OpenBB Copilot took to arrive at the
+answer:
 
 <div style={{display: 'flex', justifyContent: 'center'}}>
-<img src="https://openbb-assets.s3.amazonaws.com/docs/copilot/intermediate_result_artifact_chat_example.jpg" alt="intermediate result artifact openbb copilot" width="80%" height="80%" />
```
Comment (mnicstruwig) [inline content/terminal/openbb-copilot.md:201]: Remove from S3 after merge

File: content/terminal/openbb-copilot.md [hunk 14]
```diff
@@ -18,202 +18,318 @@ import ReactPlayer from 'react-player'
...
+When accessing data (whether from widgets on your dashboard, files you have
+uploaded, or external data sources), OpenBB Copilot will always cite the data
+source it referenced when responding to your query:
 
-Each time OpenBB Copilot generates a response, it will have a "Create widget from text" or "Create widget from table" button. Clicking this button will create a new widget on your dashboard based on the response.
+<div style={{display: 'flex', justifyContent: 'center'}}>
+<img src="https://openbb-assets.s3.amazonaws.com/docs/copilot/citations.png" alt="citations openbb copilot" width="80%" height="80%" />
+</div>
 
-| ![widget generate button openbb copilot](https://openbb-assets.s3.amazonaws.com/docs/copilot/text_widget_creation_button.jpg) | ![widget generate button openbb copilot](https://openbb-assets.s3.amazonaws.com/docs/copilot/text_widget_creation.jpg) | ![widget generate button openbb copilot](https://openbb-assets.s3.amazonaws.com/docs/copilot/text_widget_created.jpg) |
```
Comment (mnicstruwig) [inline content/terminal/openbb-copilot.md:208]: Remove from S3 after merge

File: content/terminal/openbb-copilot.md [hunk 15]
```diff
@@ -18,202 +18,318 @@ import ReactPlayer from 'react-player'
...
-|:-----------------------------------------------------------------------------------------:|:-----------------------------------------------------------------------------------------:|:-----------------------------------------------------------------------------------------:|
+If the widget that was cited by OpenBB Copilot is present on your dashboard,
+hovering over the citation will give you the option to scroll to that widget.
 
-| ![widget generate button openbb copilot](https://openbb-assets.s3.amazonaws.com/docs/copilot/table_widget_creation_button.jpg) | ![widget generate button openbb copilot](https://openbb-assets.s3.amazonaws.com/docs/copilot/table_widget_creation.jpg) | ![widget generate button openbb copilot](https://openbb-assets.s3.amazonaws.com/docs/copilot/table_widget_created.jpg) |
```
Comment (mnicstruwig) [inline content/terminal/openbb-copilot.md:211]: Remove from S3 after merge

File: content/terminal/openbb-copilot.md [hunk 16]
```diff
@@ -5,215 +5,340 @@ description: Learn how to use OpenBB Copilot to interact with the OpenBB Termina
...
 </div>
 
+
 :::note
 
-When uploading files directly to OpenBB Copilot, we use filenames to determine if a file is relevant to your query. For best results, use descriptive filenames.
-For example, if you have a technical report from Tesla released in 2024, a good filename would be `tsla_technical_report_2024.pdf`.
+When uploading files directly to OpenBB Copilot, we use filenames to determine
+if a file is relevant to your query. For best results, use descriptive
+filenames.  For example, if you have a technical report from Tesla released in
```
Comment (IgorWounds) [inline content/terminal/openbb-copilot.md]: Double white space

**Review by IgorWounds — APPROVED** [review]

File: content/terminal/openbb-copilot.md [hunk 17]
```diff
@@ -5,215 +5,340 @@ description: Learn how to use OpenBB Copilot to interact with the OpenBB Termina
...
+your questions and provide more relevant and personalized assistance throughout
+your interaction.
 
 For example, if you ask about the price-to-earnings (P/E) ratio, and then follow
 up by asking about other important financial ratios, OpenBB Copilot will
-remember the context and suggest relevant ratios:
+understand the context and suggest relevant ratios:
 
 <div style={{display: 'flex', justifyContent: 'center'}}>
 <img src="https://openbb-assets.s3.amazonaws.com/docs/copilot/pe_ratio_chat_example.jpg" alt="chat history openbb copilot" width="40%" height="40%" />
```
Comment (montezdesousa) [inline content/terminal/openbb-copilot.md]: Slightly outdated screenshot

image

File: content/terminal/openbb-copilot.md [hunk 18]
```diff
@@ -5,215 +5,340 @@ description: Learn how to use OpenBB Copilot to interact with the OpenBB Termina
...
-source it referenced when responding to your query.
+<div style={{display: 'flex', justifyContent: 'center'}}>
+<img src="https://openbb-assets.s3.amazonaws.com/docs/copilot/table_artifact.png" alt="table artifact openbb copilot" width="60%" height="60%" />
+</div>
+
+This behaviour works across all structured data.
+
+### Converting artifacts to widgets
+Every artifact that OpenBB Copilot generates can be converted to a widget. To do
+this, click the "Create widget from text" or "Create widget from table" button:
```
Comment (montezdesousa) [inline content/terminal/openbb-copilot.md]: Here the screenshot actually says "Create widget from chart", so we can be more general like:

"...click the "Create widget from [text, table or chart]" button:"

File: content/terminal/openbb-copilot.md [hunk 19]
```diff
@@ -5,215 +5,340 @@ description: Learn how to use OpenBB Copilot to interact with the OpenBB Termina
...
+
+## Citations
 
-Each time OpenBB Copilot generates a response, it will have a "Create widget from text" or "Create widget from table" button. Clicking this button will create a new widget on your dashboard based on the response.
+When accessing data (whether from widgets on your dashboard, files you have
+uploaded, or external data sources), OpenBB Copilot will always cite the data
+source it referenced when responding to your query:
+
+<div style={{display: 'flex', justifyContent: 'center'}}>
+<img src="https://openbb-assets.s3.amazonaws.com/docs/copilot/citations.png" alt="citations openbb copilot" width="80%" height="80%" />
```
Comment (montezdesousa) [inline content/terminal/openbb-copilot.md]: I'm getting the `alt`, this image is not available.

Screenshot 2024-12-04 at 11 10 19

File: content/terminal/openbb-copilot.md (hunk 18 above)
Comment (mnicstruwig) [reply content/terminal/openbb-copilot.md]: Good catch!

File: content/terminal/openbb-copilot.md (hunk 19 above)
Comment (mnicstruwig) [reply content/terminal/openbb-copilot.md]: Good catch! Let me check.

File: content/terminal/openbb-copilot.md (hunk 17 above)
Comment (mnicstruwig) [reply content/terminal/openbb-copilot.md]: It also shows localhost -- which isn't good! I'll update. Nice catch.

File: content/terminal/openbb-copilot.md (hunk 19 above)
Comment (mnicstruwig) [reply content/terminal/openbb-copilot.md]: Fixed.

File: content/terminal/openbb-copilot.md (hunk 17 above)
Comment (mnicstruwig) [reply content/terminal/openbb-copilot.md]: Fixed!

**Pull request merged**